Documentation/proposals/202405-agent-daemonset.md

```diff
@@ -72,7 +72,52 @@ The current [PrometheusAgent CRD](https://prometheus-operator.dev/docs/platform/
 * Shard
 * Storage
 
-We will add a new `mode` field that accepts either `StatefulSet` or `DaemonSet`, with `StatefulSet` being the default. If the DaemonSet mode is activated (`mode: DaemonSet`), all the unrelated fields listed above will not be accepted. In the MVP, we will simply fail the reconciliation if any of those fields are set. We will prevent users to directly switch from a live StatefulSet setup to DaemonSet, because that might break their workload if they forget to unset the unsupported fields. Following up, we will leverage validation rules with [Kubernetes' Common Expression Language (CEL)](https://kubernetes.io/docs/reference/using-api/cel/). Only then, we will allow switching from a live StatefulSet setup to DaemonSet. We already have an issue for CEL [here](https://github.com/prometheus-operator/prometheus-operator/issues/5079).
+We will add a new `mode` field that accepts either `StatefulSet` or `DaemonSet`, with `StatefulSet` being the default. If the DaemonSet mode is activated (`mode: DaemonSet`), all the unrelated fields listed above will not be accepted. In the MVP, we will simply fail the reconciliation if any of those fields are set. We will prevent users to directly switch from a live StatefulSet setup to DaemonSet, because that might break their workload if they forget to unset the unsupported fields.
+
+#### 6.1.1 CEL Validation rules
+
+To prevent users from accidentally providing invalid configurations, we will implement validation rules using [Kubernetes' Common Expression Language (CEL)](https://kubernetes.io/docs/reference/using-api/cel/).
+
+These CEL rules will affect the DaemonSet mode and prevent:
+- Setting replicas 
+- Setting storage configuration 
+- Using ServiceMonitorSelector 
+- Setting shards 
+- Setting persistentVolumeClaimRetentionPolicy 
+- Using serviceMonitorNamespaceSelector 
+- Setting serviceName 
```

Comment:
hmm... I am not sure if this is unacceptable for a Daemonset. I think it is a valid use case to have a governing service that exposes all the Daemonsets.
CC: @simonpasquier @slashpai